packages/react/src/hooks/useDate/useDate.ts

```diff
@@ -79,63 +80,91 @@ export default function useDate() {
         } else if ('string' === typeof date.$date) {
           computedDate = parseDate(date.$date);
         }
-
-        return computedDate.isValid() ? computedDate.fromNow() : '';
+        return computedDate;
       } catch (error) {
         console.error(error);
-        return '';
       }
+      return computedDate;
     },
     [currentLanguage, parseDate],
   );
 
-  const formatDate = useCallback(
-    (date: CoreDate, format = 'short'): string => {
-      let computedDate: Dayjs = dayjs();
+  const formatTimeAgo = useCallback(
+    (date: CoreDate | NumberDate): string => {
+      const computedDate = toComputedDate(date);
 
-      try {
-        if ('undefined' === typeof date) {
-          return '';
-        } else if ('string' === typeof date) {
-          computedDate = parseDate(date);
-        } else if ('number' === typeof date) {
-          computedDate = dayjs(date).locale(currentLanguage as string);
-        } else if ('number' === typeof date.$date) {
-          computedDate = dayjs(new Date(date.$date)).locale(
-            currentLanguage as string,
-          );
-        } else if ('string' === typeof date.$date) {
-          computedDate = parseDate(date.$date);
-        }
+      if (!computedDate?.isValid()) return '';
 
-        let dayjsFormat = '';
-        switch (format) {
-          case 'short':
-            dayjsFormat = 'L';
-            break;
-          case 'long':
-            dayjsFormat = 'LL';
-            break;
-          case 'abbr':
-            dayjsFormat = 'll';
-            break;
-          default:
-            dayjsFormat = format;
+      const now = dayjs();
+
+      if (computedDate.isSame(now, 'date')) {
+        if (now.diff(computedDate, 'hours') <= 3) {
+          return computedDate.fromNow();
+        } else {
+          // format HH:mm
+          return computedDate.format(t('date.format.currentDay'));
         }
+      }
 
-        return computedDate.isValid()
-          ? computedDate.locale(currentLanguage as string).format(dayjsFormat)
-          : '';
-      } catch (error) {
-        console.error(error);
-        return '';
+      if (computedDate.isSame(now.subtract(1, 'day'), 'date')) {
+        // format "Yesterday"
+        return t('date.format.yesterday');
+      }
+
+      if (now.diff(computedDate, 'days') <= 7) {
+        // format dddd
+        return computedDate.format(t('date.format.currentWeek'));
+      }
+
+      if (computedDate.isSame(now, 'year')) {
+        // format D MMM
+        return computedDate.format(t('date.format.currentYear'));
+      }
+
+      // format D MMM YYYY
+      return computedDate.format(t('date.formatpreviousYear'));
```

Comment:
```suggestion
      return computedDate.format(t('date.format.previousYear'));
```
Il manque un point ??